Tasks/PublishSymbols/PublishSymbols.ps1

```diff
@@ -31,47 +31,15 @@ try {
         Write-Host "Get-SymStorePath: $symstorePath"
     }
 
-    [string]$SymbolServerType = Get-VstsInput -Name 'SymbolServerType' -Require
     [bool]$DetailedLog = Get-VstsInput -Name 'DetailedLog' -AsBool
-    
-    if ($SymbolServerType -eq "FileShare") {
-        # Get common inputs.
-        [int]$SymbolsMaximumWaitTime = Get-VstsInput -Name 'SymbolsMaximumWaitTime' -Default '0' -AsInt
-        [timespan]$SymbolsMaximumWaitTime = if ($SymbolsMaximumWaitTime -gt 0) { [timespan]::FromMinutes($SymbolsMaximumWaitTime) } else { [timespan]::FromHours(2) }
-
-        # Unpublish symbols.
-        if ([bool]$Delete = Get-VstsInput -Name 'Delete' -AsBool) {
-            # Construct the semaphore message.
-            $utcNow = (Get-Date).ToUniversalTime()
-            $semaphoreMessage = "Unpublish: True, Machine: $env:ComputerName, BuildUri: $env:Build_BuildUri, BuildNumber: $env:Build_BuildNumber, RepositoryName: $env:Build_Repository_Name, RepositoryUri: $env:Build_Repository_Uri, Team Project: $env:System_TeamProject, CollectionUri: $env:System_TeamFoundationCollectionUri at $utcNow UTC"
-
-            # Delete the symbol store transaction.
-            [string]$SymbolsPath = Get-VstsInput -Name 'SymbolsPath' -Require
-            [string]$TransactionId = Get-VstsInput -Name 'TransactionId' -Require
-
-            Import-Module -Name $PSScriptRoot\PublishHelpers\PublishHelpers.psm1
-            Invoke-UnpublishSymbols -Share $SymbolsPath -TransactionId $TransactionId -MaximumWaitTime $SymbolsMaximumWaitTime -SemaphoreMessage $SemaphoreMessage
-            return
-        }
-
-        # Get the inputs.
-        [string]$SymbolsPath = Get-VstsInput -Name 'SymbolsPath'
-
-        if ([string]$SourceFolder = (Get-VstsInput -Name 'SourceFolder') -and
-            $SourceFolder -ne (Get-VstsTaskVariable -Name 'Build.SourcesDirectory' -Require)) {
-            Write-Warning (Get-VstsLocString -Key SourceFolderDeprecated0 -ArgumentList $SourceFolder)
-        }
-
-        [string]$SymbolsProduct = Get-VstsInput -Name 'SymbolsProduct' -Default (Get-VstsTaskVariable -Name 'Build.DefinitionName' -Require)
-        [string]$SymbolsVersion = Get-VstsInput -Name 'SymbolsVersion' -Default (Get-VstsTaskVariable -Name 'Build.BuildNumber' -Require)
-        [string]$SymbolsArtifactName = Get-VstsInput -Name 'SymbolsArtifactName'
-    }
     [bool]$SkipIndexing = -not (Get-VstsInput -Name 'IndexSources' -AsBool)
+    [bool]$NeedsPublishSymbols = Get-VstsInput -Name 'PublishSymbols' -Require -AsBool
     [bool]$TreatNotIndexedAsWarning = Get-VstsInput -Name 'TreatNotIndexedAsWarning' -AsBool
     [string]$defaultSymbolFolder = (Get-VstsTaskVariable -Name 'Build.SourcesDirectory' -Default "")
     [string]$SymbolsFolder = Get-VstsInput -Name 'SymbolsFolder' -Default $defaultSymbolFolder
-
-    if ( ($SymbolServerType -eq "FileShare") -or ($SymbolServerType -eq "TeamServices") -or (-not $SkipIndexing) ) {
```

Comment:
>($SymbolServerType -eq "FileShare") -or ($SymbolServerType -eq "TeamServices") [](start = 9, length = 78)

The $fileList is used below to publish file, 
and will cause a regression ($fileList will be empty) when $SkipIndexing = true and publishing